src/main/scala/com/microsoft/hyperspace/index/IndexLogEntry.scala

```diff
@@ -323,9 +323,22 @@ case class LogicalPlanFingerprint(properties: LogicalPlanFingerprint.Properties)
   val kind = "LogicalPlan"
 }
 object LogicalPlanFingerprint {
-  case class Properties(signatures: Seq[Signature])
+  case class Properties(signatures: Seq[Signature], planSignature: String)
```

Comment:
@imback82 @pirz @apoorvedave1 
I wonder this is good approach or no.  We need to keep `planSignature` separately for signature calculation at refreshing  - though only logical relation plan is allowed for now.

Reply:
Hmm, this seems a bit weird. :)  Can you point me where this is being used? I couldn't find.

One way to get around (depending on the use case), we can get rid of `IndexSignatureProvider` and just use two separate signatures: `FileBasedSignatureProvider` and `PlanSignatureProvider`. And we can have a utils class that combines the signature values on the rule path.

Reply:
@imback82  Outdated but for use case: https://github.com/sezruby/hyperspace/blob/metaupdate/src/main/scala/com/microsoft/hyperspace/actions/RefreshAppendAction.scala#L100
https://github.com/sezruby/hyperspace/blob/metaupdate/src/main/scala/com/microsoft/hyperspace/actions/CreateActionBase.scala#L70

`signatures` defined as `Seq[Signature]` for multiple relations. So do you suggest `Seq[Seq[Signature]]`?

This is for quick refresh, so no use case for now. Can I create `RefreshQuickAction.scala` for this w/o exposing API?

Reply:
> `signatures` defined as `Seq[Signature]` for multiple relations. So do you suggest `Seq[Seq[Signature]]`?

Does it make sense to move `fingerprint: LogicalPlanFingerprint` from `SparkPlan` to `Relation` so that the fingerprint always goes with the corresponding relation. (And we can break up `IndexSignatureProvider`). WDYT?

Reply:
@apoorvedave1 / @pirz Any suggestions?

Reply:
As a source plan can contain multiple relations, it might be better to keep plan signature in SparkPlan.
But I think it's good to calculate "FileBasedSignatureProvider" for each relation respectively, in terms of design flexibility.

Reply:
> As a source plan can contain multiple relations, it might be better to keep plan signature in SparkPlan.

Can't we have additional signature at `SparkPlan` in that case?

Can you explain the use case at a high level (instead of code), so we can better understand what we need to achieve?

Reply:
Refresh Quick mode
- collect appendedFiles, deletedFiles, latestSignature 
- latestSignature can be calculated by planSignature value + updated file based signature value
- current behavior at refresh (full and incremental):
  - use reconstructed `df` to get planSignature

Seems we could use the reconstructed `df` in the same way of the other refresh modes.

Reply:
I updated this PR w/o planSignature change and added simple RefreshQuickAction impl for use case.

Reply:
@imback82 @priz @apoorvedave1 Could you give any quick comment on this? 

BTW how about adding `stats: Map[String, String]` in CoveringIndex.properties that @imback82 mentioned in the previous discussion for `hasLineage` properties? it's just to avoid breaking changes in the next release expecting more statistics / usability changes.

Reply:
Thanks @sezruby for `RefreshQuickAction`. The change looks good to me. I left a comment whether we can use `LogicalPlanFingerprint`. If it works out, could you separate out `RefreshQuickAction` to a separate PR to better track?

> BTW how about adding `stats: Map[String, String]` in CoveringIndex.properties that @imback82 mentioned in the previous discussion for `hasLineage` config? it's just to avoid breaking changes in the next release expecting more statistics / usability changes.

There may be an option to update `ObjectMapper` to allow adding new fields while maintaining compatibility? I am also fine with adding one now. Would this be a "stat" or more like "properties"?

Reply:
> There may be an option to update ObjectMapper to allow adding new fields while maintaining compatibility? I am also fine with adding one now. 

Seems no problem to add new fields later. Let's handle it later and wrap up this release 👍

> Would this be a "stat" or more like "properties"?

I suggested `stats` as it will be added under `CoveringIndex.Properties`. How about `extra` or `aux` ..?